### DISCUSSION TITLE
Kiali pod keeps restarting after setting reconcile annotation

### DISCUSSION THREAD

**Discussion by m8-t** [kiali/kiali#7863] [Q&A]
Hey,

I ran into an issue where kiali did not discover newly added istio namespaces.  The docs state that Kiali won't check that by default and there are multiple ways to achieve this. I annotated the kiali CR to let the Operator run the Ansible Playbook again, this worked and the newly added namespaces were discovered by kiali. As I do not want to always think about annotating the kiali CR when a new istio-enabled namespace is added to the mesh I decided to use the mentioned annotation from the ansible sdk:

> You can ask the Kiali Operator to periodically reconcile the Kiali CR on a fixed schedule. See the [Ansible Operator SDK documentation describing the reconcile-period annotation](https://sdk.operatorframework.io/docs/building-operators/ansible/reference/advanced_options/#ansiblesdkoperatorframeworkioreconcile-period-custom-resource-annotation). In short, you can have the Kiali Operator periodically reconcile a Kiali CR by setting the ansible.sdk.operatorframework.io/reconcile-period annotation on the Kiali CR. For example, to reconcile this Kiail CR every 60 seconds:

```
metadata:
  kind: Kiali
  annotations:
    ansible.sdk.operatorframework.io/reconcile-period: 60s
```

I added this annotation with a reconcile-period of 4 hours, but since then the kiali pod (not the operator pod) is restarting kiali randomly like any few minutes. I cannot see any logs from the kiali operator running ansible again. But the last run showed 8 changes, which I think is like the newly added namespaces, but since then the operator kept quiet.

I can see something like this in kiali CR:

```
Name:         kiali
Namespace:    istio-system

Annotations:  ansible.sdk.operatorframework.io/reconcile-period: 4h
              kiali.io/reconcile: Thu Oct 24 05:23:45 PM CEST 2024 <<-- this was my manually annotated reconcile annotation
....
Status:
  Conditions:
    Last Transition Time:  2024-10-07T14:31:50Z
    Message:
    Reason:
    Status:                False
    Type:                  Failure
    Ansible Result:
      Changed:             8
      Completion:          2024-10-24T15:24:13.363589
      Failures:            0
      Ok:                  92
      Skipped:             52
    Last Transition Time:  2024-10-07T14:31:18Z
    Message:               Awaiting next reconciliation
    Reason:                Successful
    Status:                True
    Type:                  Running
    Last Transition Time:  2024-10-24T15:24:13Z
    Message:               Last reconciliation succeeded
    Reason:                Successful
    Status:                True
    Type:                  Successful
```

no failures here and I dont get why kiali keeps restarting since the addition of the reconcile-period.

Any hints?

**Comment by jmazzitelli** [comment]
When new namespaces are detected and added to the configmap's `accessible_namespaces` (I assume you are using 1.x and not 2.x - but I think it doesn't really matter - because in 2.x the new namespaces are just added to the configmap's `discovery_selectors` instead), that will cause the Kiali pod to restart (the operator will be the one that does that). But that should only happen upon reconciliation (which happens either (a) during the run that was kicked off when the reconcile period expires or (b) when someone touches the Kiali CR manually).

Kiali pod should not restart every few minutes. If it does, and if the operator caused it, you would see logs in the Kiali operator because that happens during a reconciliation run. I can't explain why the Kiali pod would restart every few minutes unless kubernetes itself is restarting it for some reason (resource quotas??)

The only other time this would happen that I can think of is if you turned on namespace watching - in which case, as soon as the namespace is created, the operator will kick off a reconciliation run. See number 4 here in this doc: https://kiali.io/docs/configuration/namespace-management/#operator-namespace-watching

**Comment by m8-t** [comment]
Thanks for your reply.
I use kiali 1.89.7 and use `accessible_namespaces` via regex:

```
  deployment:
    accessible_namespaces:
    - istio-system
    - someprefix-.*
  ```

I checked for events but didn't see anything important that causes the restart, resource quotas are set on the deployment, not namespace but that shouldn't exceed the limits atm, not much load there and I could see that in the events somewhere. This really started flipping after I set the reconcile-period annotation. Right now the pod is running for 5 hours, Operator logs states last reconcile 08:26, kiali pod didn't restart as expected because there were no changes. Really weird behaviour, I will keep looking but for now it looks like it works as expected again.

**Comment by m8-t** [comment]
The problem disappeared, not sure what the pod caused to reboot all the time. It is now stable, closing this issue.
